pallets/avn-anchor/src/lib.rs

```diff
@@ -55,18 +49,38 @@ pub mod pallet {
         ChainHandlerRegistered(ChainId, T::AccountId),
         /// A chain handler was updated. [chain_id, new_handler_account_id]
         ChainHandlerUpdated(ChainId, T::AccountId),
+        //// A new checkpoint was submitted. [chain_id, checkpoint_id, checkpoint]
+        CheckpointSubmitted(ChainId, CheckpointId, H256),
     }
 
     #[pallet::error]
     pub enum Error<T> {
         HandlerAlreadyExists,
         HandlerNotRegistered,
+        UnauthorizedHandler,
     }
 
     #[pallet::storage]
     #[pallet::getter(fn chain_handlers)]
     pub type ChainHandlers<T: Config> = StorageMap<_, Blake2_128Concat, ChainId, T::AccountId>;
 
+    #[pallet::storage]
+    #[pallet::getter(fn checkpoints)]
+    pub type Checkpoints<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        ChainId,
+        Blake2_128Concat,
+        CheckpointId,
+        H256,
+        ValueQuery,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn next_checkpoint_id)]
+    pub type NextCheckpointId<T: Config> =
+        StorageMap<_, Blake2_128Concat, ChainId, CheckpointId, ValueQuery>;
```

Comment:
why do we need a unique checkpoint_id per chain? Can it not be a universal nonce for simplicity?